api/client/commands.go

```diff
@@ -139,10 +143,42 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		if _, err := os.Stat(root); err != nil {
 			return err
 		}
-		filename := path.Join(root, "Dockerfile")
+
+		absRoot, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+
+		var filename string       // path to Dockerfile
+		var origDockerfile string // used for error msg
+
+		if *dockerfileName == "" {
+			// No -f/--file was specified so use the default
+			origDockerfile = "Dockerfile"
+			*dockerfileName = origDockerfile
+			filename = path.Join(absRoot, *dockerfileName)
+		} else {
+			origDockerfile = *dockerfileName
+			if filename, err = filepath.Abs(*dockerfileName); err != nil {
+				return err
+			}
+		}
+
+		if !strings.HasSuffix(absRoot, "/") {
```

Comment:
Why this checks done for default `Dockerfile` too? Shouldn't this be in non-default name condition?